### DISCUSSION TITLE
Trouble when use jit in class

### DISCUSSION THREAD

**Discussion by yiminghwang** [jax-ml/jax#12353] [Q&A] [answered]
Hi, there,

I am going to use jit to speed up my function in a class, sample code as shown in the following,
```python
class NN:
    def __init__(self, var):
         self.var = var

    @partial(jit, static_argnums=(0,))
    def func(self, params):
        # some processing codes in here
        return output
```
where var is a constant array,  the func is dependent not only on params but also on var(it will change when I do other things). if I use @partial(jit, static_argnums=(0,)), it might fix the self, thus no matter what the var is, the grad(func)(params) is always the same. How can I fit this?

**Comment by mattjj** [comment, marked as answer]
Thanks for the question! Check out this [FAQ entry](https://jax.readthedocs.io/en/latest/faq.html#how-to-use-jit-with-methods). I suggest Strategy 1.

**Reply by yiminghwang** [reply]
Thank you! it works for me!
Btw, I have an extended question,
what about the following situation?
```python
class NN:
    def __init__(self, var: List):
         self.var = var

    def func(self, params):
        # some processing codes in here
        return _func(self.var, params)

@partial(jit, static_argnums=(0,))
def _func(var: List, params)
     # some processing codes in here
     return output
```
when the var is a list(Any), it raise the TypeError: unhashable type: 'list', and the non-hashable static argument are not support.

**Reply by mattjj** [reply]
Do you need to make it static at all?

**Reply by yiminghwang** [reply]
for now, yes. Then I try to convert the list to tuple, it works.
